docker-images/operator/Makefile

```diff
@@ -0,0 +1,32 @@
+PROJECT_NAME=operator
+
+clean:
+	rm -rf lib
+	rm -rf tmp
+	rm .*.mk
+
+.topic-operator.mk: ../../topic-operator/target/*-dist.zip
```

Comment:
I wonder a bit what do these constructs do ... could you explain it please?

Reply:
The way `make` is "meant" to be used is to track the dependencies between files and rebuild outputs when the inputs have been changed. In other words in a "traditional" `Makefile` the targets are often files and the dependencies are the files which the recipe will use to upgrade the target file. `make` will avoid running a recipe if the target has a more recent `mtime` than the inputs.

What I'm doing in this `Makefile` is using these temporary files to leverage this feature of `make`. In this case these temporary files aren't outputs we care about, but we're using them only to avoid doing unnecessary work. It doesn't make much difference to build times here, but ultimately I would like to change the `Makefiles` we use in the Java modules and for the docker directories too so that I can use `make all` at the top level and rather than blindly rebuilding everything it only rebuilds those images it really needs to. (I realise you use a different workflow where you `cd` around the sources building just those parts which need to be rebuilt following a code change, but I would prefer not let `make` keep track of that for me).

Reply:
I suppose I should use a `.tmp` extension rather than `.mk`, which is confusing because it's usually used for Makefiles themselves.